From: Alexandros Frantzis <alexandros.frantzis@collabora.com>
Subject: [PATCH] win32u: Fix NtUserCallHooks return value.
Message-Id: <20220228145446.558153-1-alexandros.frantzis@collabora.com>
Date: Mon, 28 Feb 2022 16:54:46 +0200

Return the actual result value of NtUserCallHooks invocations, which
also fixes an inadvertent switch statement case fallthrough.

Signed-off-by: Alexandros Frantzis <alexandros.frantzis@collabora.com>
---
 dlls/win32u/sysparams.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/dlls/win32u/sysparams.c b/dlls/win32u/sysparams.c
index b5d82748c71..cd4ff91255e 100644
--- a/dlls/win32u/sysparams.c
+++ b/dlls/win32u/sysparams.c
@@ -4582,7 +4582,8 @@ ULONG_PTR WINAPI NtUserCallOneParam( ULONG_PTR arg, ULONG code )
     case NtUserCallHooks:
         {
             const struct win_hook_params *params = (struct win_hook_params *)arg;
-            call_hooks( params->id, params->code, params->wparam, params->lparam, params->next_unicode );
+            return call_hooks( params->id, params->code, params->wparam, params->lparam,
+                               params->next_unicode );
         }
     case NtUserFlushWindowSurfaces:
         flush_window_surfaces( arg );

-- 
2.34.1

